frontend/public/components/app.jsx

```diff
@@ -202,13 +204,15 @@ class App_ extends React.PureComponent {
   }
 }
 
-initConsolePlugins(store);
+const AppWithExtensions = withTranslation()((props) => {
+  const [contextProviderExtensions] = useResolvedExtensions(isContextProvider);
 
-const App = withExtensions({ contextProviderExtensions: isContextProvider })(App_);
```

Comment:
Can we add a 2nd param to `withExtensions` which if `true` will resolve all extensions with `useResolvedExtensions` ?

cc @vojtechszocs

Reply:
hooks can't be conditional? @christianvogt

Reply:
@glekner `withExtensions` is [HOC](https://reactjs.org/docs/higher-order-components.html) counterpart to `useExtensions` hook.

Long-term expectations:
- `useExtensions` hook gets replaced by `useResolvedExtensions` hook
- `withExtensions` HOC gets removed by refactoring related class components into functional ones

@christianvogt In the short-term, yes - I can post a PR for that; however, the 2nd HOC param needs to have an impact on `TExtensionProps` type param, similar to `ResolvedExtension` type used by `useResolvedExtensions` hook.

Reply:
@vojtechszocs i'm fine with a `withResolvedExtensions` solution as well. I think the typing would be easier.

@vojtechszocs not sure I agree with this statement.
> useExtensions hook gets replaced by useResolvedExtensions hook
Why would we remove `useExtensions`? It serves a different purpose.

I also think having an HOC is valid. It also will be useful as part of the SDK.
> withExtensions HOC gets removed by refactoring related class components into functional ones

Reply:
@christianvogt 

Regarding the removal of `useExtensions` hook - I thought we want all code to use `useResolvedExtensions` hook as the official API to consume Console extensions as part of React render flow? (Once we upgrade to TypeScript 4.x we can use [variadic tuple types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#variadic-tuple-types) feature to add `options` arg to `useResolvedExtensions` hook in order to customize its behavior.)

Regarding the removal of `withExtensions` HOC - I thought the long-term strategy is moving away from React class components in favor of functional components & hooks?

Reply:
We don't always want to resolve extension coderefs. It's up to the consumer when to resolve them. Sometimes we want to use the code ref with `React.lazy` or do something else lazily instead of resolving up front.

Reply:
Right, the `React.lazy` is a valid use case, agreed.